app/jobs/send_gpo_code_expiration_notices_job.rb

```diff
@@ -0,0 +1,83 @@
+class SendGpoCodeExpirationNoticesJob < ApplicationJob
+  queue_as :low
+
+  def initialize(analytics: nil)
+    @analytics = analytics
+  end
+
+  def perform
+    codes_to_send_notifications_for.find_each do |code|
+      user = code.profile.user
+
+      user.send_email_to_all_addresses(:gpo_code_expired, code_sent_at: code.code_sent_at)
```

Comment:
As I commented on the ticket, we are assuming here that the config value `max_mail_events_window_in_days`, currently set to 30, will always be less than the code expiration, currently set to 31. I still think that we should check that assumption somewhere so we're not accidentally sending users emails inviting them to restart IdV if they're rate limited in the future when someone changes the config value.

Reply:
I added an additional condition, `user_is_not_rate_limited_for_gpo` in 072f4f4fcddb6afcaafd487fb136cbff6f9da521 (and fixed a bug in it in ab6c8769521ded538259989ed56546425c06eb57). The idea is that we just don't email folks who are currently rate limited for GPO.